### PR TITLE
Document some properties of NavigationMesh

### PR DESCRIPTION
**Pull request by skyace65**
Documents some properties of the NavigationMesh. I'd like to do more but there's only so much I can figure out without a working project to mess around with.

### CODE REVIEW

File: doc/classes/NavigationMesh.xml [hunk 1]
```diff
@@ -104,18 +105,23 @@
...
 		</member>
 		<member name="geometry/source_geometry_mode" type="int" setter="set_source_geometry_mode" getter="get_source_geometry_mode" default="0">
+			Which geometry is used to generate the [NavigationMesh].
 		</member>
 		<member name="geometry/source_group_name" type="StringName" setter="set_source_group_name" getter="get_source_group_name">
+			The name of the group that is used to generate the [NavigationMesh].
 		</member>
 		<member name="polygon/verts_per_poly" type="float" setter="set_verts_per_poly" getter="get_verts_per_poly" default="6.0">
 		</member>
 		<member name="region/merge_size" type="float" setter="set_region_merge_size" getter="get_region_merge_size" default="20.0">
```
Comment (AndreaCatania) [inline doc/classes/NavigationMesh.xml:121]: Max distance, between two different adjacent regions edges, within - those are considered welded.

@Calinou please make it more easy to understand.

File: doc/classes/NavigationMesh.xml [hunk 2]
```diff
@@ -104,18 +105,23 @@
...
 		<member name="geometry/parsed_geometry_type" type="int" setter="set_parsed_geometry_type" getter="get_parsed_geometry_type" default="0">
+			What kind of geomerty is used to generate the [NavigationMesh].
 		</member>
 		<member name="geometry/source_geometry_mode" type="int" setter="set_source_geometry_mode" getter="get_source_geometry_mode" default="0">
+			Which geometry is used to generate the [NavigationMesh].
 		</member>
 		<member name="geometry/source_group_name" type="StringName" setter="set_source_group_name" getter="get_source_group_name">
+			The name of the group that is used to generate the [NavigationMesh].
 		</member>
 		<member name="polygon/verts_per_poly" type="float" setter="set_verts_per_poly" getter="get_verts_per_poly" default="6.0">
```
Comment (AndreaCatania) [inline doc/classes/NavigationMesh.xml:119]: Count of vertices per each poly: a high number results in a lower resolution but much better performance.

File: doc/classes/NavigationMesh.xml (hunk 1 above)
Comment (Calinou) [reply doc/classes/NavigationMesh.xml:121]: @AndreaCatania I have very little experience with navigation in Godot, so I'm afraid I can't help much here.

File: doc/classes/NavigationMesh.xml (hunk 2 above)
Comment (Calinou) [reply doc/classes/NavigationMesh.xml:119]: > The number of vertices to use per polygon. Higher values will improve performance at the cost of lower precision.

**Comment by skyace65** [comment]
I've added Calinous descriptions for both of those properties. Also I don't know if I made this clear or not in the description but I'm fine with this being merged as is, even though not everything is documented.

File: doc/classes/NavigationMesh.xml (hunk 1 above)
Comment (AndreaCatania) [reply doc/classes/NavigationMesh.xml:122]: Yep perfect!

File: doc/classes/NavigationMesh.xml (hunk 2 above)
Comment (AndreaCatania) [reply doc/classes/NavigationMesh.xml:119]: Perfect!

**Review by AndreaCatania — APPROVED** [review]

**Comment by AndreaCatania** [comment]
I've already marked it as approved, feel free to also fill the not documented fields with the suggested description.

**Comment by skyace65** [comment]
Added a description for the properties agent max height and agent max slope.

**Comment by akien-mga** [comment]
Superseded by #41912.

**Pull request closed without merging**